### PR TITLE
clear abandoned match unit tests

### PR DESCRIPTION
**Pull request by bernardoaraujor**
adds unit tests to `clear_abandoned_match`

I had to refine the implementation of `Match::clear_abandoned_bet()` because I found an edge case where if the bets were too small, the janitor incentive was smaller than the `minimum_balance`. In that case, janitor gets the full prize now.

### CODE REVIEW

**Review by lana-shanghai — CHANGES_REQUESTED** [review]

File: src/lib.rs [hunk 1]
```diff
@@ -155,26 +156,33 @@ pub mod pallet {
...
 				false,
 			)?;
 			Ok(())
 		}
+
+		pub fn janitor_incentive(&self) -> (BalanceOf<T>, BalanceOf<T>) {
+			let winner_prize = self.bet_amount.saturating_add(self.bet_amount);
+			let (janitor_incentive, actual_prize) = match Percent::from_percent(T::IncentiveShare::get()) * winner_prize {
+				i if i >= T::Assets::minimum_balance(self.bet_asset_id) => (i, winner_prize - i),
+				_ => (winner_prize, 0u32.into()), // if the bet is too small, we just give the whole prize to the janitor
```
[the remark this answers is not in the document]
Comment (lana-shanghai) [reply src/lib.rs]: Essentially the janitor can get the maximum of two values: incentive or two minimum values. Something like: 
`cmp::max(T::Assets::minimum_balance(self.bet_asset_id) * 2, janitor_incentive)`

**Review by lana-shanghai — APPROVED** [review]
LGTM

**Review by lana-shanghai — APPROVED** [review]
